SleepEngine/Engine/Utils/MacroUtils.h

```diff
@@ -80,5 +80,26 @@ CONST_REF_SETTER(type, setterName, member)
 
 #pragma endregion
 
+#define COMPONENT_SHORTCUTS(componentType, shortcutName)\
+inline componentType& shortcutName(Object* obj) \
+{   \
+    return *obj->getComponent<componentType>();  \
+}   \
+    \
+inline componentType* shortcutName##Ptr(Object* obj)  \
+{   \
+    return obj->getComponent<componentType>();  \
+}   \
+    \
+inline componentType& shortcutName(Object& obj)  \
+{   \
+    return *obj.getComponent<componentType>();  \
+}   \
+    \
+inline componentType* shortcutName##Ptr(Object& obj)  \
+{   \
+    return obj.getComponent<componentType>();  \
+}   \
+
```

Comment:
ohhhhh shit

Reply:
ok I like it

Reply:
what about variant with const object reference?

Reply:
added, also refactored the copypaste